vkcs/dc/resource_ip_port_forwarding.go

```diff
@@ -0,0 +1,376 @@
+				Optional:    true,
+				Description: "Port selector.",
+			},
+
+			"to_port": schema.Int64Attribute{
+				Optional:    true,
+				Description: "Destination port selector.",
+			},
+
+			"name": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "Name of the conntrack helper",
+			},
+
+			"description": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "Description of the conntrack helper",
+			},
+
+			"created_at": schema.StringAttribute{
+				Computed:    true,
+				Description: "Creation timestamp",
+			},
+
+			"updated_at": schema.StringAttribute{
+				Computed:    true,
+				Description: "Update timestamp",
+			},
+
+			"region": schema.StringAttribute{
+				Optional:    true,
+				Computed:    true,
+				Description: "The `region` to fetch availability zones from, defaults to the provider's `region`.",
+			},
+		},
+		Description: "Manages a direct connect ip port forwarding resource._note_ This resource requires Sprut SDN to be enabled in your project.",
+	}
+}
+
+func (r *IPPortForwardingResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+	r.config = req.ProviderData.(clients.Config)
+}
+
+func (r *IPPortForwardingResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data IPPortForwardingResourceModel
+	diags := req.Plan.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	region := data.Region.ValueString()
+	if region == "" {
+		region = r.config.GetRegion()
+	}
+
+	networkingClient, err := r.config.NetworkingV2Client(region, networking.SprutSDN)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating VKCS networking client", err.Error())
+		return
+	}
+
+	ipPortForwardingCreateOpts := ipportforwardings.CreateOpts{
+		Name:          data.Name.ValueString(),
+		Description:   data.Description.ValueString(),
+		DCInterfaceID: data.DCInterfaceID.ValueString(),
+		Protocol:      data.Protocol.ValueString(),
+		Source:        data.Source.ValueString(),
+		Destination:   data.Destination.ValueString(),
+		Port:          int(data.Port.ValueInt64()),
+		ToDestination: data.ToDestination.ValueString(),
+		ToPort:        int(data.Port.ValueInt64()),
+	}
+
+	ipPortForwardingResp, err := ipportforwardings.Create(networkingClient, &ipportforwardings.IPPortForwardingCreate{IPPortForwarding: &ipPortForwardingCreateOpts}).Extract()
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating vkcs_dc_ip_port_forwarding", err.Error())
+		return
+	}
+	ipPortForwardingID := ipPortForwardingResp.ID
+	resp.State.SetAttribute(ctx, path.Root("id"), ipPortForwardingID)
+
+	data.ID = types.StringValue(ipPortForwardingID)
+	data.Region = types.StringValue(region)
+	data.DCInterfaceID = types.StringValue(ipPortForwardingResp.DCInterfaceID)
```

Comment:
btw - do we really need to do this since tf calls read function right after this one returns ok?

Reply:
Yes, we need this. After create step, terraform checks that all values are known, otherwise we' ll get 
> "Error: Provider returned invalid result object after apply"